puppet/lib/puppet/util/puppetdb/config.rb

```diff
@@ -14,6 +14,8 @@ def self.load(config_file = nil)
         :soft_write_failure        => false,
```

Comment:
seems like min_successful_writes and soft_write_failure should be mutually exclusive with some sort of config error